packages/design-system/src/Button/Button.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { forwardRef } from "react";
+import { useButton } from "@react-aria/button";
+import { AriaButtonProps } from "@react-types/button";
+import clsx from "classnames";
+
+import { StyledButton, ButtonContent } from "./Button.styles";
+import { ButtonProps } from "./Button.types";
+import { useDOMRef } from "Hooks/useDomRef";
+import { Icon } from "Icon";
+import {
+  ButtonClassName,
+  ButtonLoadingClassName,
+  ButtonLoadingIconClassName,
+  ButtonContentClassName,
+  ButtonContentChildrenClassName,
+  ButtonContentIconStartClassName,
+  ButtonContentIconEndClassName,
+} from "./Button.constants";
+import { Spinner } from "Spinner";
+
+const Button = forwardRef<HTMLButtonElement, ButtonProps>(
+  (props, ref): JSX.Element => {
+    const {
+      as,
+      children,
+      className,
+      endIcon,
+      isDisabled,
+      isLoading,
+      kind,
+      size,
+      startIcon,
+      UNSAFE_height,
+      UNSAFE_width,
+      ...rest
+    } = props;
+    const buttonRef = useDOMRef(ref);
+    const { buttonProps } = useButton(
+      {
+        ...rest,
+        elementType: as,
+      } as AriaButtonProps,
+      buttonRef,
+    );
+
+    return (
+      <StyledButton
+        as={as || "button"}
+        {...buttonProps}
+        UNSAFE_height={UNSAFE_height}
+        UNSAFE_width={UNSAFE_width}
+        className={clsx(ButtonClassName, className)}
+        disabled={isDisabled}
```

Comment:
We should also disable the button when isLoading is try - `disabled={isDisabled || isLoading}`

Reply:
I would argue that. Loading and Disable UI can be different. If they are same in our design, then ok.

Reply:
@yaldram  yeah our designs are different that's why we are not doing this.